### PR TITLE
Hide `Variations` section when it is empty

### LINKED ISSUE #36181: [Enhancement]: Variations section is visible even when there are no variations to show
The `Variations` section is visible even when there aren't any to list. Before creating any option, the variations section shouldn't be visible.
Now the Variations section is being shown with an infinite spinner.

Screenshot 2022-12-26 at 09 35 56

### PR DESCRIPTION
**Pull request by octaedro**
### All Submissions:

-   [x] Have you followed the [WooCommerce Contributing guideline](https://github.com/woocommerce/woocommerce/blob/trunk/.github/CONTRIBUTING.md)?
-   [x] Does your code follow the [WordPress' coding standards](https://make.wordpress.org/core/handbook/best-practices/coding-standards/)?
-   [x] Have you checked to ensure there aren't other open [Pull Requests](https://github.com/woocommerce/woocommerce/pulls) for the same update/change?

### Changes proposed in this Pull Request:

Hide the `Variations` section when there aren't any variations to show. 

Closes #36181.

- [ ] This PR is a very minor change/addition and does not require testing instructions (if checked you can ignore/remove the next section).

### How to test the changes in this Pull Request:

1. Navigate to the new product management experience (Products -> Add new (MVP))
2. Click the "Options" tab
3. The `Variations` section should be hidden.
4. Go to a product with variations `wp-admin/admin.php?page=wc-admin&path=/product/{product_id}`.
5. Click the "Options" tab and verify that the Variations section is being shown correctly. 

### Other information:

-   [ ] Have you added an explanation of what your changes do and why you'd like us to include them?
-   [ ] Have you written new tests for your changes, as applicable?
-   [ ] Have you created a changelog file for each project being changed, ie `pnpm --filter=<project> changelog add`?

### FOR PR REVIEWER ONLY:

-   [ ] I have reviewed that everything is sanitized/escaped appropriately for any SQL or XSS injection possibilities. I made sure Linting is not ignored or disabled.

### CODE REVIEW

**Review by joshuatf — COMMENTED** [review]
Thanks for creating this issue and tackling it, Fernando!

I'm still seeing the variations section shown with this PR checked out.  I think we might need check for the existence of variations from the data store inside the variations section.
Screen Shot 2022-12-27 at 4 58 12 PM

**Comment by octaedro** [comment]
Thank you @joshuatf for reviewing this PR.
I addressed the suggestion you made in [your comment](https://github.com/woocommerce/woocommerce/pull/36202#pullrequestreview-1231143234). Could you take another look at this PR?

File: plugins/woocommerce-admin/client/products/sections/product-variations-section.tsx [hunk 1]
```diff
@@ -12,6 +17,29 @@ import { ProductSectionLayout } from '../layout/product-section-layout';
 import { Variations } from '../fields/variations';
 
 export const ProductVariationsSection: React.FC = () => {
+	const { values } = useFormContext< Product >();
+	const productId = values.id;
+
+	const { totalCount } = useSelect(
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/sections/product-variations-section.tsx]: This is working well for me on the default state, but when options are changed by deleting them if they already existed or by adding them on a new product, the variations section does not change.

We might need cache invalidation for this store when options are changed or maybe we should just use the same `getProductAttributes` request that is used in options and check that there is at least 1 item.

**Review by joshuatf — COMMENTED** [review]
Thanks for the update!  Looks good for each case of options existing or not, but ran into a little bit of trouble when changing the options.  

Could you check if that's happening on your end? Left a couple possible solutions in the comments.

File: plugins/woocommerce-admin/client/products/sections/product-variations-section.tsx [hunk 2]
```diff
@@ -12,6 +18,44 @@ import { ProductSectionLayout } from '../layout/product-section-layout';
...
+			};
+			return {
+				totalCount:
+					getProductVariationsTotalCount< number >( requestParams ),
+			};
+		},
+		[ productId, options ]
+	);
+
+	if ( options.length === 0 || totalCount === 0 || isNaN( totalCount ) ) {
```
Comment (joshuatf) [inline plugins/woocommerce-admin/client/products/sections/product-variations-section.tsx]: I think we can remove all of this `totalCount` logic now and just rely on the `options.length`.

On a new product page when adding options, this `totalCount` will not get updated and the variations will remain hidden.

**Review by joshuatf — COMMENTED** [review]
Thanks for the updated logic!

I left one case where things are broken on a new product page, but left a comment on how we can fix it.  Let me know if that works and we can approve if so.

File: plugins/woocommerce-admin/client/products/sections/product-variations-section.tsx (hunk 2 above)
Comment (octaedro) [reply plugins/woocommerce-admin/client/products/sections/product-variations-section.tsx]: Thank you @joshuatf for your review.

This is interesting; when I place a breakpoint [here](https://github.com/woocommerce/woocommerce/pull/36202/files/d88d91cfda8661b7fe5e1b0ac64c34b72f56dc86#diff-f0bcdfc13aba51d445b30531587b77a929f1e47ca619b9cd4d53e87df0595735R41) the `totalCount`const is loaded correctly after adding a new option. I guess that there is something related to the variations creation time.

As you mentioned, it seems like we can remove the `totalCount`. I removed that logic in the commit [e9d7f4d](https://github.com/woocommerce/woocommerce/pull/36202/commits/e9d7f4d232b6f1c1fb6f8a051002c14d53c39c42)

**Review by joshuatf — APPROVED** [review]
Testing well on my end and LGTM!  🚢

**Pull request merged**